src/instruction.rs

```diff
@@ -1278,6 +1469,44 @@ impl Instruction {
         }
     }
 
+    /// Return immutable references to the [`Qubit`]s contained within an instruction
+    /// TODO: replace the logic in Program::get_used_qubits with this
+    #[allow(dead_code)]
+    pub(crate) fn get_qubits(&self) -> Vec<&Qubit> {
+        match self {
+            Instruction::Gate(gate) => gate.qubits.iter().collect(),
+            Instruction::Measurement(measurement) => vec![&measurement.qubit],
+            Instruction::Reset(reset) => match &reset.qubit {
+                Some(qubit) => vec![qubit],
+                None => vec![],
+            },
+            Instruction::Delay(delay) => delay.qubits.iter().collect(),
+            Instruction::Fence(fence) => fence.qubits.iter().collect(),
+            Instruction::Capture(capture) => capture.frame.qubits.iter().collect(),
+            Instruction::Pulse(pulse) => pulse.frame.qubits.iter().collect(),
+            Instruction::RawCapture(raw_capture) => raw_capture.frame.qubits.iter().collect(),
+            _ => vec![],
```

Comment:
`pyQuil` supports using `QubitPlaceholder`s in `Pragma`s. That looks to be missing here, presumably because the `Pragma` instruction in `quil-rs` doesn't distinguish a qubit from any other kind of argument. I'm not sure what `Pragma`s are associated with qubits, and how useful they are, but is that something we should try to port back here?

Reply:
Yeah, that's a mistake, per para 9.2 in the [Quil spec](https://quil-lang.github.io/), so I'd rather not try to coerce it in here. It should be a breaking change in pyQuil v4. Do you think that'll impact many use cases?